File: bill-payment-system/src/main/java/com/cognizant/billpaymentsystem/service/PaymentService.java
/**
 * 
 */
package com.cognizant.billpaymentsystem.service;

import java.util.List;

import com.cognizant.billpaymentsystem.model.Bill;
import com.cognizant.billpaymentsystem.model.Payment;

/**
 * @author 810512
 *
 */
public interface PaymentService {
	
	Bill getPaymentDetails(String vendorName,String username,String billPaymentGateway);
	List<Bill> getPaymentDetailsOfUser(String vendorName,String username);
	void doPayment(Payment payment);
}

File: webapp/src/app/payment/payment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { AuthenticateService } from '../authenticate.service';
import { Payment } from '../payment.model';

@Injectable({
  providedIn: 'root'
})
export class PaymentService {

  token:string='';
  constructor(private httpClient:HttpClient,private authService:AuthenticateService) { }

  getPaymentDetailsByVendorAndUserAndPaymentGateway(vendorName:string,username:string,billPaymentGateway:string){
    this.token=this.authService.accessToken;
    let headers=new HttpHeaders();
    headers=headers.set('Authorization','Bearer '+this.token);
    return this.httpClient.get(`http://localhost:8083/payment/${vendorName}/${username}/${billPaymentGateway}`,{headers})
  }

  getPaymentDetailsByVendorAndUser(vendorName:string,username:string){
    this.token=this.authService.accessToken;
    let headers=new HttpHeaders();
    headers=headers.set('Authorization','Bearer '+this.token);
    return this.httpClient.get(`http://localhost:8083/payment/${vendorName}/${username}`,{headers})
  }

  doPayment(payment:Payment){
    this.token=this.authService.accessToken;
    let headers=new HttpHeaders();
    headers=headers.set('Authorization','Bearer '+this.token);
    return this.httpClient.post<any>(`http://localhost:8083/payment`,payment,{headers});
  }

}
